### ISSUE TITLE
Add a config to PRs where we assume there is only 1 GPU available

### ISSUE THREAD

**Issue by janeyx99** [pytorch/pytorch#66511]
We recently had a gap in PR coverage where we did not catch when a test case attempted to access an invalid GPU from this PR https://github.com/pytorch/pytorch/pull/65914. We should capture that in PR testing somehow to catch these early next time.

Action:
Make our tests run on only one "available" GPU.
We could take advantage of this and split up our tests to run on separate GPUs when they're available!

cc @seemethere @malfet @pytorch/pytorch-dev-infra @mruberry

**Comment by malfet** [comment]
Can we split it into two issues:
 - Add `CUDA_VISIBLE_DEVICES=0` for all non-multigpu tests? (Although, it this a good thing always?)
 - Create another task for running tests in parallel on multi-gpu runner

**Comment by mruberry** [comment]
> Can we split it into two issues:
> 
> * Add `CUDA_VISIBLE_DEVICES=0` for all non-multigpu tests? (Although, it this a good thing always?)

This is not necessarily a pure improvement because another config we've wanted to run on historically (@ngimel likely recalls the details better than I do) was 2 GPU configs but where the tests run on the non-default GPU. 

The code to set the primary CUDA test device is here:

https://github.com/pytorch/pytorch/blob/53e6aca8b3672877fea15ba8b1566c204ba2951f/torch/testing/_internal/common_device_type.py#L470
